Sources/NIO/SelectorEpoll.swift

```diff
@@ -93,6 +93,35 @@ extension SelectorEventSet {
     }
 }
 
+@usableFromInline struct EPollUserData {
+    // We're backing this by a tuple to get standard C layout
+    @usableFromInline var registrationID: SelectorRegistrationID
+    @usableFromInline var fileDescriptor: CInt
+
+
+    @inlinable init(registrationID: SelectorRegistrationID, fileDescriptor: CInt) {
+        assert(MemoryLayout<UInt64>.size == MemoryLayout<EPollUserData>.size)
+        self.registrationID = registrationID
+        self.fileDescriptor = fileDescriptor
+    }
+
+    @inlinable init(rawValue: UInt64) {
+        let registrationID = UInt32(truncatingIfNeeded: (rawValue & (UInt64(UInt32.max) << 32)) >> 32)
+        let fileDescriptor = CInt(truncatingIfNeeded: rawValue & UInt64(UInt32.max))
+
+        self = .init(registrationID: SelectorRegistrationID(rawValue: registrationID),
+                     fileDescriptor: fileDescriptor)
+    }
+}
+
+extension UInt64 {
+    @inlinable
+    init(_ epollUserData: EPollUserData) {
```

Comment:
nit: I think having the `epollUserData` label here would actually be helpful

Reply:
The [Swift API Guidelines](https://swift.org/documentation/api-design-guidelines/) say

>  In initializers that perform value preserving type conversions, omit the first argument label, e.g. Int64(someUInt32)